File: src/RNetPi.Core/Packets/PacketS2CSourceDescriptiveText.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x35
/// Source Descriptive Text
/// Data:
///     (Unsigned Char) SourceID
///     (Short) Time
///     (String) Text
/// </summary>
public class PacketS2CSourceDescriptiveText : PacketS2C
{
    public const byte ID = 0x35;

    public PacketS2CSourceDescriptiveText(byte sourceID, ushort time, string text)
    {
        Writer.Write(sourceID);
        Writer.Write(time);
        WriteNullTerminatedString(text);
    }

    public override byte GetID() => ID;
}

File: src_js/server/packets/PacketS2CSourceDescriptiveText.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x35
 * Source Descriptive Text
 * Data:
 *      (Unsigned Char) SourceID
 *      (Short) Time
 *      (String) Text
 */
class PacketS2CSourceDescriptiveText extends PacketS2C {
    constructor(sourceID, time, text) {
        super();

        this._buffer.writeUInt8(sourceID);
        this._buffer.writeUInt16LE(time);
        this._buffer.writeStringNT(text);
    }

    getID() {
        return 0x35;
    }
}

module.exports = PacketS2CSourceDescriptiveText;
